packages/jsii-rosetta/lib/jsii/assemblies.ts

```diff
@@ -19,28 +19,32 @@ export interface LoadedAssembly {
 /**
  * Load assemblies by filename or directory
  */
-export async function loadAssemblies(assemblyLocations: string[]) {
-  const ret: LoadedAssembly[] = [];
-  for (const loc of assemblyLocations) {
-    const stat = await fs.stat(loc); // eslint-disable-line no-await-in-loop
+export async function loadAssemblies(
+  assemblyLocations: readonly string[],
+  validateAssemblies: boolean,
+): Promise<readonly LoadedAssembly[]> {
+  return Promise.all(assemblyLocations.map(loadAssembly));
+
+  async function loadAssembly(location: string): Promise<LoadedAssembly> {
+    const stat = await fs.stat(location);
     if (stat.isDirectory()) {
-      ret.push({
-        assembly: await loadAssemblyFromFile(path.join(loc, '.jsii')), // eslint-disable-line no-await-in-loop
-        directory: loc,
-      });
-    } else {
-      ret.push({
-        assembly: await loadAssemblyFromFile(loc), // eslint-disable-line no-await-in-loop
-        directory: path.dirname(loc),
-      });
+      return loadAssembly(path.join(location, '.jsii'));
     }
+    return {
+      assembly: await loadAssemblyFromFile(location, validateAssemblies),
+      directory: path.dirname(location),
+    };
   }
```

Comment:
Re-worked this code largely to remove the linter exclusions... It does the same work, except in parallel instead of serially, which could be a little faster, since this involves I/O wait.